src/pages/index.js

```diff
@@ -59,54 +32,126 @@ const config = {
   inputErrorClass: "modal__input_type_error",
   errorClass: "modal__error_visible",
 };
+const profileButton = document.querySelector(".profile__button");
+
+const profileAvatarModal = document.querySelector("#profile-avatar-modal");
+const profileImage = document.querySelector(".profile__image");
+const profileAvatarModalForm = profileAvatarModal.querySelector(".modal__form");
+
+const userInfo = new UserInfo(".profile__title", ".profile__description");
+const cardSection = new Section(renderCard, cardListEl);
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "f1dab6ed-e5ba-44ef-bef1-7369d7e9bf0d",
+    "Content-Type": "application/json",
+  },
+});
+
+api.getUserInfo().then((res) => {
+  userInfo.setUserInfo(res);
+});
+api.getInitialCards().then((res) => {
+  cardSection.renderItems(res);
+});
+api.getUserInfo().then((res) => {
+  profileImage.src = res.avatar;
+});
+
+const cardDeleteModal = new PopupQuestion("#delete-card-modal");
 
 const editFormValidator = new FormValidator(config, profileEditForm);
 editFormValidator.enableValidation();
 const addFormValidator = new FormValidator(config, addCardFormElement);
 addFormValidator.enableValidation();
+const profileAvatarFormValidator = new FormValidator(
+  config,
+  profileAvatarModalForm
+);
 
+profileAvatarFormValidator.enableValidation();
 const imagePopup = new PopupWithImage("#image-modal");
 imagePopup.setEventListeners();
 
-const cardPopup = new PopupWithForm("#add-card-modal", handleAddCardFormSubmit);
+const cardPopup = new PopupWithForm(
+  "#add-card-modal",
+  // handleAddCardFormSubmit
+  (link, submitButton) => {
+    api.addCard(link, submitButton).then((res) => {
+      cardSection.addItem(res);
+    });
+    addFormValidator.toggleButtonState();
+    cardPopup.close();
+  }
+);
 cardPopup.setEventListeners();
 
-const userInfo = new UserInfo(".profile__title", ".profile__description");
+const profilePopup = new PopupWithForm(
+  "#profile-avatar-modal",
+  (url, submitButton) => {
+    // handles the avatar picture submit
+    api.setUserAvatar(url, submitButton).then((res) => {
+      profileImage.src = res.avatar;
+    });
+
+    profileAvatarFormValidator.toggleButtonState();
+    profilePopup.close();
+  }
+);
+profilePopup.setEventListeners();
+
 const userInfoPopup = new PopupWithForm(
   "#profile-edit-modal",
-  handleProfileFormSubmit
+  (values, submitButton) => {
+    userInfo.setUserInfo(values);
```

Comment:
You need to update `DOM` only in blocks `then` after a successful response from the server so that when there is an error it wouldn’t change

**Please, correct it everywhere**